src/selectors/index.ts

```diff
@@ -0,0 +1,12 @@
+import { createSelector } from 'reselect';
+import { Comment } from '../state/comments'
+import { State } from '../state'
+
+const getComments = (state: State) => state.comments.comments;
+
+export function selectCommentsForContentPathFactory (contentPath: string) {
+    return createSelector(
+        getComments,
+        (comments: Map<number, Comment>) => [...comments.values()].filter((comment: Comment) => comment.contentPath === contentPath)
```

Comment:
**Question**: Shouldn't `(comments: Map<number, Comment>)`'s type be inferred from the `State` in `getComments`? Maybe `reselect` does not play nice with TypeScript.

Reply:
Good point! It does indeed pick it up